x-pack/filebeat/module/salesforce/logout/config/logout.yml

```diff
@@ -0,0 +1,95 @@
+{{ if eq .input "salesforce" }}
+
+type: salesforce
+enabled: true
+{{ if .api_version }}
+version: {{ .api_version }}
+{{ end }}
+auth.oauth2:
+{{ if .jwt_enabled }}
+  jwt_bearer_flow:
+    enabled: true
+    {{ if .client_id }}
+    client.id: {{ .client_id }}
+    {{ end }}
+    {{ if .client_username }}
+    client.username: {{ .client_username }}
+    {{ end }}
+    {{ if .client_key_path }}
+    client.key_path: {{ .client_key_path }}
+    {{ end }}
+    {{ if .jwt_url }}
+    url: {{ .jwt_url }}
+    {{ end }}
+{{ else }}
+  user_password_flow:
+    enabled: true
+    {{ if .client_id }}
+    client.id: {{ .client_id }}
+    {{ end }}
+    {{ if .client_secret }}
+    client.secret: {{ .client_secret }}
+    {{ end }}
+    {{ if .token_url }}
+    token_url: {{ .token_url }}
+    {{ end }}
+    {{ if .username }}
+    username: {{ .username }}
+    {{ end }}
+    {{ if .password }}
+    password: {{ .password }}
+    {{ end }}
+{{ end }}
+# Query params will be overwritten by request.transforms from start of the input and 
+# it is to indicate that this url is for Apex type of events as cursor stores the url as source.
+# Each filebeat input cursor source needs to be uniquely identified with a name.
+url: {{ .url }}
+event_monitoring_method:
+  event_log_file:
+  {{ if .event_log_file }}
+    enabled: true
+  {{ else }}
+    enabled: false
+  {{ end }}
+    interval: {{ .elf_interval }}
+    query:
+      default: "SELECT CreatedDate,LogDate,LogFile FROM EventLogFile WHERE {{ if .initial_interval }}LogDate > [[ formatTime(parseDuration(-{{ .initial_interval }})), "2006-01-02T15:04:05+0700") ]] AND{{ end }} Interval = '{{ .log_file_interval }}' AND EventType = 'Logout' ORDER BY LogDate ASC NULLS FIRST"
+      value: "SELECT CreatedDate,LogDate,LogFile FROM EventLogFile WHERE Interval = '{{ .log_file_interval }}' AND EventType = 'Logout' AND CreatedDate > [[ .cursor.event_log_file.last_event_time ]] ORDER BY LogDate ASC NULLS FIRST"
```

Comment:
What if we take event type as a var too? Then mostly all event types have the same query. Easy maintenance.

Reply:
Works for login, logout but it will not if the query is different i.e. apex has a different query.